helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -282,7 +282,8 @@ public ZKHelixManager(String clusterName, String instanceName, InstanceType inst
     // read cloud config from ZK and set cloudConfig in HelixManagerProperty
     _helixManagerProperty = helixManagerProperty;
     _helixManagerProperty.getHelixCloudProperty().populateFieldsWithCloudConfig(
-        HelixPropertyFactory.getCloudConfig(_zkAddress, _clusterName));
+        HelixPropertyFactory.getCloudConfig(_zkAddress, _clusterName,
+            helixManagerProperty.getZkConnectionConfig()));
```

Comment:
Question: could you remind me why we make the change in the first place?  What's in `helixManagerProperty.getZkConnectionConfig()`?

Reply:
helixManagerProperty.getZkConnectionConfig() returns an object of `RealmAwareZkConnectionConfig`.
I our current Helix use case, the user is defining the multiZkRouting endpoint in the `RealmAwareZkConnectionConfig`. However, the original `getCloudConfig` create a zkClient only using the multiZkRouting in java env variable. When user pass in the config but not set it in env variable, we will fail to create zkClient even when user passed in correct config.